src/main/resources/static/js/grid.js

```diff
@@ -8,22 +8,98 @@ function Cell(opt) {
     this.parent = opt.parent;
     this.rows = opt.rows;
     this.state = 0;
+    this.student;
 
     this.setState = function (state) {
         this.el.removeClass(this.states[this.state]);
         this.state = state;
         this.el.addClass(this.states[this.state]);
     };
 
-    this.el.on('click', {cell: this}, function (e) {
-        cell = e.data.cell;
-        if (cell.parent.editable) {
-            cell.el.removeClass(cell.states[cell.state]);
-            cell.state++;
-            if (!cell.states[cell.state]) {
-                cell.state = 0;
+    this.setStudent = function(student) {
+        if(student) {
+            this.student = student;
+            this.el.html(student.username);
+        }
+    };
+
+    this.removeStudent = function() {
+        if(this.student){
+            this.student = undefined;
+            this.el.html("");
+        }
+    };
+
+    this.getInfo = function() {
+        console.log(this.parent);
+        return {
+            "id": this.id,
+            "row": this.row,
+            "col": this.col,
+            "state": this.state,
+            "student": {
+                "id": this.getStudentID()
+            },
+            "course": {
+                "id": this.parent.id
+            }
+        }
+    };
+
+    this.setID = function(id) {
+        this.id = id;
+    };
+
+    this.getStudentID = function() {
+        if(this.student) {
```

Comment:
this will make this function return undefined if there's no student right?

Reply:
Yeah you're right. Would it be better to return null?